layouts/DocsPage/Versions.tsx

```diff
@@ -2,30 +2,90 @@ import cn from "classnames";
 import { useRouter } from "next/router";
 import { useCallback, useEffect, useMemo, useState } from "react";
 import { Dropdown } from "components/Dropdown";
-import type { VersionsInfo } from "./types";
+import type { VersionsInfo, VersionsDropdown } from "./types";
 import styles from "./Versions.module.css";
 
-const renderVersion = (version: string) => `Version ${version}`;
+// renders strikethrough for deprecated versions
+const RenderVersion = (version: VersionsDropdown) => {
+  if (version.deprecated) return <s>Version {version.value}</s>;
+
+  if (version.value === "Older Versions") return version.value;
+  else return `Version ${version.value}`;
+};
+
+// renders the default box selection
+const pickOption = (options: VersionsDropdown[], id: string) =>
+  options.find(({ value }) => value === id);
+
+// assigns keys and values based on the value prop
+const pickId = ({ value }: VersionsDropdown) => value;
 
 const Versions = ({
   current,
   available,
   disabled,
   className,
   getNewVersionPath,
+  latest,
 }: VersionsInfo) => {
   const router = useRouter();
   const [currentItem, setCurrentItem] = useState<string>(current);
-  const versions = useMemo(() => [...available].reverse(), [available]);
 
+  const latestNumber: number = Math.floor(+latest);
+  const validVersion = useCallback(
+    (num: number) => {
+      return num >= latestNumber - 2 ? true : false;
+    },
+    [latestNumber]
+  );
```

Comment:
This can probably be simplified (the `useCallback` can be removed). If the function is restructured to take in two parameters (`num` and `latestVersion`), it can be hoisted outside of `Versions` so it's just a helper utility.